### PR TITLE
Fix issue where newly published rc's cannot be fetched with full version string

### PR DESCRIPTION
**Pull request by jdmundrawala**
For examples, the following was not working:
http://localhost:9393/metadata?v=12.4.0.rc.0&prerelease=false&nightlies=false&p=el&pv=6&m=x86_64

This commit also replaces all the versioning code with mixlib-versioning

This is one way to fix it

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -312,20 +313,21 @@ class InvalidDownloadPath < StandardError; end
...
+      #Opscode::Version::GitDescribe,
+      #Opscode::Version::OpscodeSemVer,
+      #Opscode::Version::Incomplete
+    #].each do |version|
+      #begin
+        #break v = version.new(version_string)
+      #rescue
+        #nil
+      #end
+    #end
```
Comment (lamont-granquist) [inline app.rb]: delete all the comments and you'll get even better red/green statistics.

Comment (schisamo) [inline app.rb]: Can we rip these comments out?

**Comment by lamont-granquist** [comment]
It reads fine to me, but I can't tell you if there's any subtle differences between it and mixlib-versioning...

**Comment by schisamo** [comment]
:+1:

**Comment by jdmundrawala** [comment]
seems likely that there are going to be subtle differences, but the tests pass, so yolo?

**Comment by lamont-granquist** [comment]
[IYOLO](http://www.youtube.com/watch?v=z5Otla5157c)

**Comment by btm** [comment]
yay @jdmundrawala :shipit:

**Pull request merged**